docs/README-ock.rst

```diff
@@ -0,0 +1,123 @@
+Opencryptoki
+============
+
+The PKCS#11 implementation `opencryptoki` can be used with the
+pkcs11-provider. This document describes a basic installation and setup for
+testing purposes with the pkcs11-provider.
+
+Installation as package
+=======================
+
+Install opencryptoki from the distribution's repository.
+
+RedHat/Fedora
+-------------
+
+Install the base packages (including depending packages):
+
+    sudo dnf install opencryptoki
+
+For the pkcs11-provider testing environment, the software token is required.
+Other token libraries can be installed as well.
+
+    sudo dnf install opencryptoki-swtok
+
+The location of the default configuration for opencryptoki is
+`/etc/opencryptoki/opencryptoki.conf`
+
+Debian/Ubuntu
+-------------
+
+Install the base packages (including depending packages):
+
+    sudo apt install opencryptoki
+
+The location of the default configuration for opencryptoki is
+`/etc/opencryptoki/opencryptoki.conf`
+
+Installation from source
+========================
+
+Get the source from github, build and install.
+		
+    git clone https://github.com/opencryptoki/opencryptoki.git
+    cd opencryptoki
+    ./bootstrap
+    ./configure
+    make
+    sudo make install
+    sudo ldconfig
+
+The location of the default configuration for opencryptoki is
+`/usr/local/etc/opencryptoki/opencryptoki.conf`
+
+Configuration
+=============
+
+For the pkcs11-provider tests, the default configuration should be
+sufficient. If the opencryptoki configuration has changed, please run the
+initialization script once.
+
+    sudo scripts/ock-once-swtok.sh
+
+The initialization script will only work with the swtok. If another token
+should be used, it must be initialized manually.
+
+The initialization script uses a default user pin. If another pin should be
+used, please specify it in the environment variable OCK_PIN.
+
+    sudo OCK_PIN=11223344 scripts/ock-once-swtok.sh
+
+Setup
+=====
+
+The pkcs11-provider tests require an openssl configuration and a set of keys
+in the token for testing purposes. The setup script will prepare all these.
+
+    export OPENSSL_CONF=tests/openssl.cnf
+    scripts/ock-setup.sh
+
```

Comment:
the ocp-setup script should generate the appropriate openssl conf file, not the other way around, I think.

Reply:
The `tests/openssl.cnf` is generated by autotools. This is ok for the replacement of the provider library path, but if we have multiple Cryptoki libraries (e.g. nss and opencryptoki), each Cryptoki test setup needs its own openssl configuration file. Should we create separate files for each Cryptoki configuration (e.g. openssl-nss..cnf, openssl-ock.cnf) during the configure step?

Reply:
Yes I think configure.ac should find out which softotkens are available, and for each of them, if available, we'll generate a different openssl.cnf file.